tests/api2/plugin.py

```diff
@@ -0,0 +1,379 @@
+
+import pytest
+import sys
+import os
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from auto_config import pool_name
+from functions import GET, POST, DELETE, wait_on_job
+
+JOB_ID = None
+job_results = None
+not_freenas = GET("/system/is_freenas/").json() is False
+reason = "System is not FreeNAS skip Jails test"
+to_skip = pytest.mark.skipif(not_freenas, reason=reason)
+
+# default URL
+repos_url = 'https://github.com/freenas/iocage-ix-plugins.git'
+index_url = 'https://raw.githubusercontent.com/freenas/iocage-ix-plugins/master/INDEX'
+plugin_index = GET(index_url).json()
+plugin_list = list(plugin_index.keys())
+
+# custom URL
+repos_url2 = 'https://github.com/ericbsd/iocage-ix-plugins.git'
+index_url2 = 'https://raw.githubusercontent.com/ericbsd/iocage-ix-plugins/11.3-RELEASE/INDEX'
+plugin_index2 = GET(index_url2).json()
+plugin_list2 = list(plugin_index.keys())
+
+plugin_objects = [
+    "id",
+    "state",
+    "type",
+    "release",
+    "plugin_repository"
+]
+
+
+@to_skip
+def test_01_activate_jail_pool():
+    results = POST('/jail/activate/', pool_name)
+    assert results.status_code == 200, results.text
+    assert results.json() is True, results.text
+
+
+@to_skip
+def test_02_verify_jail_pool():
+    results = GET('/jail/get_activated_pool/')
+    assert results.status_code == 200, results.text
+    assert results.json() == pool_name, results.text
+
+
+@to_skip
+def test_03_get_list_of_installed_plugin():
+    results = GET('/plugin/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+
+
+@to_skip
+def test_04_verify_plugin_repos_is_in_official_repositories():
+    results = GET('/plugin/official_repositories/')
+    assert repos_url in results.json(), results.text
+
+
+@to_skip
+def test_05_get_list_of_available_plugins_job_id():
+    global JOB_ID
+    payload = {
+        "plugin_repository": repos_url
+    }
+    results = POST('/plugin/available/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), int), results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_06_verify_list_of_available_plugins_job_id_is_successfull():
+    global job_results
+    job_status = wait_on_job(JOB_ID)
+    assert job_status['state'] == 'SUCCESS', job_status['results']
+    job_results = job_status['results']
+
+
+@to_skip
+@pytest.mark.parametrize('plugin', plugin_list)
+def test_07_verify_available_plugin_(plugin):
+    assert isinstance(job_results['result'], list), str(job_results)
+    for plugin_info in job_results['result']:
+        if plugin in plugin_info['plugin']:
+            assert plugin in plugin_info['plugin'], str(job_results)
```

Comment:
This and the assert below will always be `true`. We should do something like the following
```
assert plugin in [p['plugin'] for p in job_results['result']]
```

Reply:
And skip the for loop altogether